lms/resources/lti_launch.py

```diff
@@ -99,9 +99,8 @@ def custom_canvas_api_domain(self):
         return self._request.parsed_params.get("custom_canvas_api_domain")
 
     @property
-    def should_use_section_groups(self):
-        """Return True if section groups rather than a course group should be used."""
-        if not self.is_canvas:
-            return False
+    def canvas_sections_enabled(self):
+        """Return True if Canvas sections is enabled for this request."""
+        ai_getter = self._request.find_service(name="ai_getter")
 
-        return self._request.feature("section_groups")
+        return self.is_canvas and ai_getter.canvas_sections_enabled()
```

Comment:
During an LTI launch request we use `context.canvas_sections_enabled`. This ensures that even if the `"section_groups"` feature flag is on and the application instance has a Canvas developer key and has `canvas_sections_enabled` on, we _still_ won't try to use sections if we're being launched by a non-Canvas LMS.

This is an edge-case: a user would have to give us a Canvas developer key and secret on the `/welcome` form and then use the LTI consumer key and secret that we generate for them in a _non-Canvas_ LMS. But it could happen, and if we tried to enable sections the results would be really confusing for both the user and for us/support.